azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -398,7 +398,7 @@ func resourceAzureRMCosmosDBAccountFailoverPolicyHash(v interface{}) int {
 func validateAzureRmCosmosDBAccountName(v interface{}, k string) (ws []string, errors []error) {
 	value := v.(string)
 
-	r, _ := regexp.Compile("[a-z0-9-]")
+	r, _ := regexp.Compile("^([a-z0-9\\-]+)$")
```

Comment:
Running the tests it looks like we can remove the brackets but not the `^`&`$` - updated